src/v/model/record.h

```diff
@@ -227,7 +227,8 @@ class record {
     const iobuf& value() const { return _value; }
     iobuf release_value() { return std::exchange(_value, {}); }
     iobuf share_value() { return _value.share(0, _value.size_bytes()); }
-    bool has_value() const { return _val_size >= 0; }
+    bool has_value() const { return _val_size > 0; }
```

Comment:
i'm testing this with kafka and there seems to be a distinction between empty string/buffer (_val_size == 0) and tombstone (no value, _val_size = -1)

Reply:
Thanks for calling this out, will revisit changes and add more test coverage here to ensure correctness.

Reply:
Seems like we could use some basic produce tests with both a Kafka/Redpanda client/server to ensure we are considering the same records as tombstones or not. Will look into this.